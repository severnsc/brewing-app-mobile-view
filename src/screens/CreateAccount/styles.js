import { StyleSheet } from "react-native";
import { white } from "../../constants";

export default StyleSheet.create({
	container: {
		justifyContent: "center",
		alignItems: "center",
		backgroundColor: white,
		margin: 5,
		top: 100
	},
	form: {
		width: "100%",
		paddingHorizontal: 20,
		paddingVertical: 10
	},
	input: {
		marginVertical: 10
	}
});
