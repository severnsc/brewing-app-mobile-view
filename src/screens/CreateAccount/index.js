import React from "react";
import { ActivityIndicator, View, AlertIOS } from "react-native";
import PropTypes from "prop-types";
import {
  Button,
  GradientView,
  ConfirmPasswordInput,
  TextInput
} from "../../components";
import { KeyboardAvoidingView } from "react-native";
import styles from "../styles";
import { white } from "../../constants";

const CreateAccount = ({
  createAccount,
  createAccountError,
  username,
  usernameError,
  setUsername,
  usernameLoading,
  email,
  emailError,
  setEmail,
  emailLoading,
  password,
  passwordError,
  setPassword,
  confirmPassword,
  setConfirmPassword,
  createAccountLoading
}) => (
  <GradientView>
    <KeyboardAvoidingView style={styles.container}>
      <View style={styles.form}>
        {createAccountError && AlertIOS.alert("Error!", createAccountError)}
        <TextInput
          label="Username"
          testID="signupUsername"
          errorTestID="usernameInputError"
          style={styles.input}
          value={username}
          isError={!!usernameError}
          errorText={usernameError}
          loading={usernameLoading}
          onChange={setUsername}
        />
        <TextInput
          label="Email"
          testID="signupEmail"
          errorTestID="emailInputError"
          style={styles.input}
          value={email}
          isError={!!emailError}
          errorText={emailError}
          onChange={setEmail}
          loading={emailLoading}
        />
        <TextInput
          label="Password"
          testID="signupPassword"
          password={true}
          style={styles.input}
          value={password}
          isError={!!passwordError}
          errorText={passwordError}
          onChange={setPassword}
        />
        <ConfirmPasswordInput
          testID="signupConfirmPassword"
          password={password}
          value={confirmPassword}
          onChange={setConfirmPassword}
          style={styles.input}
        />
        {createAccountLoading ? (
          <ActivityIndicator />
        ) : (
          <Button
            testID="signupFormButton"
            textTestID="signupFormButtonText"
            success={createAccountLoading}
            disabled={createAccountLoading}
            textColor={white}
            value="Sign Up"
            onPress={createAccount}
            style={styles.button}
          />
        )}
      </View>
    </KeyboardAvoidingView>
  </GradientView>
);

CreateAccount.propTypes = {
  createAccount: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired,
  usernameError: PropTypes.string,
  setUsername: PropTypes.func.isRequired,
  usernameLoading: PropTypes.bool.isRequired,
  email: PropTypes.string.isRequired,
  emailError: PropTypes.string,
  setEmail: PropTypes.func.isRequired,
  emailLoading: PropTypes.bool.isRequired,
  password: PropTypes.string.isRequired,
  passwordError: PropTypes.string,
  setPassword: PropTypes.func.isRequired,
  confirmPassword: PropTypes.string.isRequired,
  setConfirmPassword: PropTypes.func.isRequired,
  createAccountLoading: PropTypes.bool.isRequired
};

CreateAccount.defaultProps = {
  createAccount: () => {},
  username: "",
  usernameError: null,
  setUsername: () => {},
  usernameLoading: false,
  email: "",
  emailError: null,
  setEmail: () => {},
  emailLoading: false,
  password: "",
  passwordError: null,
  setPassword: () => {},
  confirmPassword: "",
  setConfirmPassword: () => {},
  createAccountLoading: false
};

export default CreateAccount;
