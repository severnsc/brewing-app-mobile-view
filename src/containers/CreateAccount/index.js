import React from "react";
import PropTypes from "prop-types";
import { CREATE_USER } from "../../graphql";
import { graphql, compose } from "react-apollo";
import { CreateAccount } from "../../screens";
import { DASHBOARD } from "../../constants";
import { validateUsername } from "../../modules/validation";

const CreateAccountContainer = ({ mutate, navigation }) => {
  const createAccount = ({ username }, { email }, password, confirmPassword) =>
    mutate({
      variables: {
        user: {
          username,
          email,
          password: password.value,
          confirmPassword: confirmPassword.value
        }
      }
    }).then(({ data: { createUser } }) => {
      if (createUser.errors.length === 0) {
        navigation.navigate(DASHBOARD);
      }
      if (createUser.errors.find(err => err.location.field === "createUser")) {
        return Promise.reject();
      }
      return createUser;
    });
  return (
    <CreateAccount
      createAccount={createAccount}
      onUsernameChange={validateUsername}
    />
  );
};

CreateAccountContainer.propTypes = {
  mutate: PropTypes.func.isRequired
};

CreateAccountContainer.defaultProps = {
  mutate: () => {}
};

export default compose(graphql(CREATE_USER))(CreateAccountContainer);
