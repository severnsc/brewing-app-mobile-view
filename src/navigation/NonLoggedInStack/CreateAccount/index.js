import React from "react";
import { CreateAccount } from "../../containers";
import { Icon, Title } from "../../components";
import { white } from "../../constants";

const createAccount = {
  CREATE_ACCOUNT: {
    screen: CreateAccount,
    navigationOptions: {
      title: "Sign Up",
      headerStyle: { height: 60 },
      headerTransparent: true,
      headerTitleStyle: { color: white, fontSize: 40 },
      headerLeftContainerStyle: { paddingLeft: 10 },
      headerBackImage: () => (
        <Icon name="ios-arrow-back" size="lg" color={white} />
      ),
      headerTitle: ({ allowFontScaling, style, children }) => (
        <Title
          value={children}
          style={style}
          allowFontScaling={allowFontScaling}
          testID="createAccountTitle"
          color="white"
        />
      )
    }
  }
};

export default createAccount;
