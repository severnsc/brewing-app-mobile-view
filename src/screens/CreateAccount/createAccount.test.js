import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import CreateAccount from ".";

describe("Create Account", () => {
  describe("layout", () => {
    it("should match the snapshot", () => {
      const createAccount = renderer.create(<CreateAccount />);
      const tree = createAccount.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("usernameError prop", () => {
    it("should set the error prop on UsernameInput", () => {
      const error = "UsernameInput error";
      const createAccount = jest.fn();
      const onChange = jest.fn();
      const createAccountScreen = shallow(
        <CreateAccount createAccount={createAccount} usernameError={error} />
      );
      const form = createAccountScreen.find("Form").prop("children")(
        ["", "", "", ""],
        onChange,
        createAccount
      );
      expect(form.props.children[0].props.error).toEqual(error);
    });
  });
  describe("emailError prop", () => {
    it("should set the error prop on EmailInput", () => {
      const error = "EmailInput error";
      const createAccount = jest.fn();
      const onChange = jest.fn();
      const createAccountScreen = shallow(
        <CreateAccount createAccount={createAccount} emailError={error} />
      );
      const form = createAccountScreen.find("Form").prop("children")(
        ["", "", "", ""],
        onChange,
        createAccount
      );
      expect(form.props.children[1].props.error).toEqual(error);
    });
  });
  describe("submitting create account form", () => {
    it("should call the createAccount prop", () => {
      const createAccount = jest.fn();
      const createAccountScreen = shallow(
        <CreateAccount createAccount={createAccount} />
      );
      createAccountScreen.find("Form").simulate("submit");
      expect(createAccount).toHaveBeenCalled();
    });
  });
});
