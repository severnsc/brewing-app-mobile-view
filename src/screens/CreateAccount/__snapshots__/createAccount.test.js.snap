// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Account layout should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "margin": 5,
      "top": 100,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
    testID="signupForm"
  >
    <View>
      <View
        style={
          Object {
            "borderBottomColor": undefined,
            "borderBottomWidth": 1,
            "marginVertical": 10,
          }
        }
      >
        <Text
          accessibilityLabel="Username"
          style={
            Object {
              "color": undefined,
            }
          }
        >
          Username
        </Text>
        <TextInput
          autoFocus={false}
          keyboardType="default"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          testID="signupUsername"
          value=""
        />
      </View>
    </View>
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Email"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Email
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        testID="signupEmail"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Password"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Password
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={true}
        testID="signupPassword"
      />
    </View>
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Confirm Password"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Confirm Password
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        secureTextEntry={true}
        testID="signupConfirmPassword"
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#37c172",
          "borderColor": "#e9ecee",
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="signupFormButton"
    >
      <Text
        accessibilityLabel="Sign Up"
        style={
          Object {
            "color": "#fff",
            "fontSize": undefined,
          }
        }
        testID="signupFormButtonText"
      >
        Sign Up
      </Text>
    </View>
  </View>
</View>
`;
